subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -783,6 +787,70 @@ static int pgps_request_all(void)
 	return pgps_request(&request);
 }
 
+#if defined(CONFIG_NRF_CLOUD_PGPS_DOWNLOAD_TRANSPORT_HTTP)
+/* handle incoming P-GPS response packets */
+int nrf_cloud_pgps_process(const char *buf, size_t buf_len)
+{
+	static char host[CONFIG_DOWNLOAD_CLIENT_MAX_HOSTNAME_SIZE];
+	static char path[CONFIG_DOWNLOAD_CLIENT_MAX_FILENAME_SIZE];
+	int err;
+	struct nrf_cloud_pgps_result pgps_dl = {
+		.host = host,
+		.host_sz = sizeof(host),
+		.path = path,
+		.path_sz = sizeof(path)
+	};
+
+	if (state == PGPS_NONE) {
+		LOG_ERR("P-GPS subsystem is not initialized.");
+		return -EINVAL;
+	}
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+	LOG_HEXDUMP_DBG(buf, buf_len, "MQTT packet");
+#endif
+	if (!buf_len) {
+		LOG_ERR("Zero length packet received");
+		state = PGPS_NONE;
+		return -EINVAL;
+	}
+
+	if (ignore_packets) {
+		LOG_ERR("IGNORING PACKETS");
+		LOG_HEXDUMP_INF(buf, buf_len, "Unexpected packet");
+		return -EINVAL;
+	}
+
+	err = nrf_cloud_parse_pgps_response(buf, &pgps_dl);
+	if (err) {
+		return err;
+	}
+
+	err = nrf_cloud_pgps_begin_update();
+	if (err) {
+		return err;
+	}
+
+	int sec_tag = SEC_TAG;
+
+	if (FORCE_HTTP_DL && (strncmp(pgps_dl.host, "https", 5) == 0)) {
+		int i;
+		int len = strlen(pgps_dl.host);
```

Comment:
not sure if this is better than the loop...
`memmove (&pgps_dl.host[4], &pgps_dl.host[5], strlen(&pgps_dl.host[4]);`